src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * The TaskList class manages a list of tasks.
+ * It provides functionalities to add, delete, retrieve, and search for tasks within the list.
+ */
+public class TaskList {
+    private final ArrayList<Task> lst;
```

Comment:
Perhaps a more contextual naming could be used for your ArrayList here and initialize the field!
```suggestion
    private final ArrayList<Task> tasks = new ArrayList<>();
```